Postgresql upstream do not build postmaster binary

fila43 opened the pull request:
Postgresql upstream do not build postmaster binary anymore since postgresql 16. So This commit should replace postmaster binary with postgres binary.
Related: https://bugzilla.redhat.com/show_bug.cgi?id=2219370 More info: https://postgrespro.com/blog/pgsql/5969981#commit_81266442

zmiklank reviewed (changes requested):
Thank you for the work on this PR.

Some nitpicks:
 1) The `README.rpm-dist.template` file still contains POSTMASTER string.
 2) can you please add subject(summary) into commit message? Like [this](https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53).
 3) I grep for postman occurrence and found one in TODO file - I understand that this is not functional nor production file, however, could you please add some explanation of the postmaster service there so it will be understandable in the future?
 
 Do you plan to release new version of postgresql-setup with this change?

praiskup commented:
I miss the background story in the commit message.  There's the link `postmaster -> postgres` since ever, do we know when we started providing it?

Is it safe to expect `"$PGDATA/postgres.pid"` exists even with PG <= 16?

praiskup commented on postgresql.init.in:
```diff
@@ -4,13 +4,13 @@
-# processname: postmaster
-# pidfile: /var/run/postmaster.PORT.pid
+# processname: postgres
+# pidfile: /var/run/postgres.PORT.pid
```
I'd drop this instead of changing;  the file doesn't exist on this location nowadays.

zmiklank replied:
+1

praiskup commented:
In `configure.ac` we already check for `PGMAJORVERSION`.  We can do some additional expansion if `>= 16`, e.g. `PGSETUP_SUBST_REQ([POSTGRES_BINARY], [The PostgreSQL server binary])`.  Then you can replace `postmaster` with `@POSTGRES_BINARY@` and you stay compatible with older PG versions.

praiskup commented:
It would be nice to have a testing postgresql.src.rpm build (in Copr?) with this not-yet-merged postgresql-setup.tar.gz, I'd happily do some testing!

fila43 committed:
Postgresql upstream do not build postmaster binary any more since

postgresql 16. So This commit should replace postmaster binary by
postgres binary.
Related: https://bugzilla.redhat.com/show_bug.cgi?id=2219370
More info: https://postgrespro.com/blog/pgsql/5969981#commit_81266442

fila43 commented:
@praiskup thank you for your review. It seems that just the binary file `postmaster` is missing in the latest (16) version. But `postmaster.pid` persists. So, I would change only the binary name. This change should be safe from a compatibility point of view. Since `postgres` binary is present even in Postgresql 9.2. 

The copr build I will add ASAP .

fila43 commented:
Copr build: https://copr.fedorainfracloud.org/coprs/fjanus/postgresql-setup-8.9/build/6384370/

praiskup commented on configure.ac:
```diff
@@ -161,7 +161,7 @@ AC_ARG_WITH([scl],
             [scl_enabled_build=:])
 
-SYSTEMD_EXECSTART='${bindir}/postmaster -D $${PGDATA}'
+SYSTEMD_EXECSTART='${bindir}/postgres -D $${PGDATA}'
```
This seems like the only non-documentation change, right?  `/usr/bin/postgres` is available even in PostgreSQL 9.2 (RHEL 7).  LGTM

praiskup reviewed (approved):
Thank you!
